tests/ipam_and_scope_then_pool/main.tf

```diff
@@ -0,0 +1,57 @@
+resource "random_string" "this" {
+  length  = 8
+  upper   = false
+  special = false
+  numeric = false
+}
+
+module "create_ipam" {
+  source = "../.."
+
+  vpc_ipam = {
+    ipam = {
+      name = "default_ipam"
+      operating_regions = [
+        { region_name = "us-east-1" },
+        { region_name = "us-west-1" }
+      ]
+      cascade     = true
+      description = random_string.this.result
+    }
+    scopes = [
+      {
+        name = "high_level_container"
+        #module.create_ipam.aws_vpc_ipam.this[0]
+        ipam_id     = aws_vpc_ipam.this["default_ipam"].id
```

Comment:
I added `name` as attr of the `vpc_ipam.ipam` object. however, i can't reference it by it's name attribute. 

` Error: Reference to undeclared resource
│ 
│   on main.tf line 25, in module "create_ipam":
│   25:         ipam_id     = aws_vpc_ipam.this["default_ipam"].id
│ 
│ A managed resource "aws_vpc_ipam" "this" has not been declared in the root module.`

Reply:
Yeah, the resource is not available in this context, here in the test config. The resource is only available in the module itself.

For the scope, the module _always_ attaches it to the ipam resource that it creates. So here in the test, just remove the `ipam_id` attribute.

```suggestion
```